### PR TITLE
Benchmark {k256, s256, mcpi} opcodes using DependentCost

### PR DESCRIPTION
**Pull request by Dentosal**
Closes https://github.com/FuelLabs/fuel-core/issues/1325

### CODE REVIEW

File: benches/benches/set/crypto.rs [hunk 1]
```diff
@@ -71,41 +73,33 @@ pub fn run(c: &mut Criterion) {
...
+    let linear = super::generate_linear_costs();
+
+    let mut bench_k256 = c.benchmark_group("k256");
+    for i in &linear {
+        bench_k256.throughput(Throughput::Bytes(*i as u64));
+        run_group_ref(
+            &mut bench_k256,
+            format!("{i}"),
+            VmBench::new(op::k256(RegId::HP, RegId::ZERO, 0x10)).with_prepare_script(
+                vec![op::movi(0x11, 32), op::aloc(0x11), op::movi(0x10, 5)],
```
Comment (xgreenx) [inline benches/benches/set/crypto.rs:85]:
```suggestion
                vec![op::movi(0x11, 32), op::aloc(0x11), op::movi(0x10, *i)],
```

File: benches/benches/set/mem.rs [hunk 2]
```diff
@@ -114,16 +106,28 @@ pub fn run(c: &mut Criterion) {
...
-                op::aloc(0x11),
-                op::move_(0x10, RegId::HP),
-            ]),
-    );
+    let mut mem_mcpi = c.benchmark_group("mcp");
+    for i in &linear {
+        if *i >= (1 << 12) {
+            // Too large for immedidate value
+            continue
+        }
```
Comment (xgreenx) [inline benches/benches/set/mem.rs:114]: Hmm, maybe it is worth simply charging for the worst case with `1 << 12`=)

**Review by xgreenx — APPROVED** [review]

File: benches/benches/set/mem.rs (hunk 2 above)
Comment (Dentosal) [reply benches/benches/set/mem.rs]: Maybe. I'll still have this PR updating the benches to `DependentCost`, and change it to use a constant cost later when we have done actual benchmarking.

File: benches/benches/set/mem.rs [hunk 3]
```diff
@@ -114,16 +106,30 @@ pub fn run(c: &mut Criterion) {
...
-        &mut c.benchmark_group("mcpi"),
-        "mcpi",
-        VmBench::new(op::mcpi(0x10, RegId::ZERO, Imm12::MAX.to_u16()))
-            .with_prepare_script(vec![
-                op::movi(0x11, Imm12::MAX.to_u16() as u32),
-                op::aloc(0x11),
-                op::move_(0x10, RegId::HP),
-            ]),
-    );
+    let mut mem_mcpi = c.benchmark_group("mcp");
```
Comment (xgreenx) [inline benches/benches/set/mem.rs:109]:
```suggestion
    let mut mem_mcpi = c.benchmark_group("mcpi");
```

**Review by xgreenx — APPROVED** [review]

**Pull request merged**